rotkehlchen/api/rest.py

```diff
@@ -917,6 +917,31 @@ def user_set_premium_credentials(
         result_dict['result'] = True
         return api_response(result_dict, status_code=HTTPStatus.OK)
 
+    @require_loggedin_user()
+    def user_change_password(
+        self,
+        name: str,
+        password: str,
+        new_password: str,
+    ) -> Response:
+        result_dict: Dict[str, Any] = {'result': None, 'message': ''}
+
+        if name != self.rotkehlchen.data.username:
+            result_dict['message'] = f'Provided user "{name}" is not the logged in user'
+            return api_response(result_dict, status_code=HTTPStatus.BAD_REQUEST)
+
+        if password != self.rotkehlchen.data.password:
+            result_dict['message'] = f'Provided current password is not correct'
+            return api_response(result_dict, status_code=HTTPStatus.UNAUTHORIZED)
+
+        try:
+            self.rotkehlchen.data.change_password(new_password=new_password)
```

Comment:
Remove the commented out code below. And also if you add a return value, and it's `False` you should return an error response from the API from here.

Reply:
Should this be within the "try" block (just store the change_password response and test if it's False?) or is there an "Error" (like "InputError" used below) that I can use?

Reply:
it can be out of the try block. Try is just to catch exceptions. In that case you would not throw an exception but simply return True/False. In case of False which was due to the error of rekey then you just return an api error.